### DISCUSSION TITLE
React onUpdate() content not updated.

### DISCUSSION THREAD

**Discussion by davidbonachera** [ueberdosis/tiptap#1858] [General Questions] [answered]
Hi,

I'm trying to use the onUpdate function of the library (cf: below), but the content doesn't change when I type characters.

The onUpdate get triggered to each change in the text but the value stay the same.

Anyone with a similar issue?

```typescript
const editor = useEditor({
        onUpdate() {
            console.log(editor?.getHTML())
        },
        extensions: [
            StarterKit,
        ],
        content: `
          <p>
            Ask your question a <em>question</em>.
          </p> 
        `,
    })
```

**Comment by davidbonachera** [comment, marked as answer]
Nevermind find out by trying, that it needs the data argument to get the updated value of the editor :


```typescript
const editor = useEditor({
        onUpdate(data) {
            console.log(data.editor.getHTML())
        },
        extensions: [
            StarterKit,
        ],
        content: `
          <p>
            Ask your question a <em>basic</em> example of <strong>here</strong>.
          </p> 
        `,
    })
```

**Reply by umerjaved178** [reply]
As you can see in the attached photo that I am receiving an extra argument in **debounce** function i.e. **notesID**
**notesID** is async, it is initially undefined and then it has value let's say after 1 sec. The problem is even though notesID will get updated in the component yet I have the old value in the **onUpdate** function
How can I fix that issue

![image](https://user-images.githubusercontent.com/25152580/168451770-97b94b43-40a1-4004-9098-caa0857e1cc2.png)
